### ISSUE TITLE
"Assume yes" flag for `dvc commit`

### ISSUE THREAD

**Issue by Northo** [iterative/dvc#10499]
**tldr: add `-y`, `--yes` flag to `dvc commit`, disabling interactive [y/n] prompt.**

Consider the workflow scenario described in the [`dvc commit` docs](https://dvc.org/doc/command-reference/commit), where we want to commit after making changes to dependencies, that does not affect output:

> Often we edit source code, configuration, or other files that are specified as dependencies in [dvc.yaml](https://dvc.org/doc/user-guide/project-structure/dvcyaml-files) (deps field) in a way that doesn't cause any changes to stage outputs. For example: reformatting input data, adding code comments, etc. However, DVC notices all changes to dependencies and expects you to reproduce the corresponding pipeline ([dvc repro](https://dvc.org/doc/command-reference/repro)). You can use [dvc commit](https://dvc.org/doc/command-reference/commit) instead to force accepting these new versions without having to execute stage commands.

We have some pipelines that run the same script many times, with the script listed in the stages' deps. If we make changes to the script, which will not affect the output, it is convenient to simply do `dvc commit`, without running repro again. However, it is inconvenient to have to click through all the interactive `dependencies ['xxx.py'] of stage: 'yyy' changed. Are you sure you want to commit it? [y/n]` prompts. It would be great with an opt-in flag like `--assume-yes`, `--yes` to disable the prompt.

**Comment by Northo** [comment]
Just realized that this is what the `-f`/`--force` flag does...

My bad. If anything, might consider changing the description of the flag, as "commit data even if hash values for dependencies or outputs did not change" made me believe it was concerned with the data only, not deps. But maybe that's jsut me 🤷
